### PR TITLE
chore: bump hardhat version to decrease flakiness

### PR DESCRIPTION
**Pull request by itegulov**
## What ❔

I think the source for flakiness might be caused by https://github.com/NomicFoundation/hardhat/commit/45025fbbb3381351f9f3cf4cdaf0afd39aa3d77b so I am trying to bump hardhat to resolve

## Why ❔

Because CI is FUBAR

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.
- [ ] Linkcheck has been run via `zk linkcheck`.

### CODE REVIEW

File: etc/ERC20/package.json [hunk 1]
```diff
@@ -6,9 +6,9 @@
   "devDependencies": {
     "@matterlabs/hardhat-zksync-deploy": "^0.6.1",
     "@matterlabs/hardhat-zksync-solc": "0.4.2",
-    "hardhat": "=2.16.0"
+    "hardhat": "=2.18.2"
```
Comment (popzxc) [inline etc/ERC20/package.json:9]: image

It seems that we have hardhat defined in multiple places (and we may call it from different places via `zk`), and also it's a bit sus that `yarn.lock` wasn't affected.

Comment (itegulov) [reply etc/ERC20/package.json]: Yup, just bumped it everywhere else too just in case

Comment (popzxc) [reply etc/ERC20/package.json]: I mean, we probably should bump it in the contracts submodule as well, I guess.

Comment (itegulov) [reply etc/ERC20/package.json]: Absolutely, I am just trying to see if this change alone will fix the flakiness as the init errors are consistently coming from `etc/ERC20`

**Comment by itegulov** [comment]
Not sure how statistically significant this is but I just had three successful runs in a row (i.e. no errors during "Init", some flaky integration tests are still present though):
[Attempt 1](https://github.com/matter-labs/zksync-era/actions/runs/9014062677/attempts/1?pr=1891)
[Attempt 2](https://github.com/matter-labs/zksync-era/actions/runs/9014062677/attempts/2?pr=1891)
[Attempt 3](https://github.com/matter-labs/zksync-era/actions/runs/9014062677/job/24769658887?pr=1891)

**Review by popzxc — APPROVED** [review]

**Pull request merged**